src/tools.h

```diff
@@ -56,6 +57,23 @@ namespace zim {
 
   std::map<std::string, int> read_valuesmap(const std::string& s);
 
+  std::string readFullMimetypeAndCounterString(std::istream& in);
+  std::pair<std::string, int> parseASingleMimetypeCounter(const std::string& s);
+
+  template<class Filter>
+  entry_index_type countMimeType(const std::string& counterData, Filter filter) {
+    entry_index_type count = 0;
+    std::istringstream datastream(counterData);
+    while (datastream) {
+      auto mtcStr = readFullMimetypeAndCounterString(datastream);
+      auto mtc = parseASingleMimetypeCounter(mtcStr);
+      if (!mtc.first.empty() && filter(mtc.first)) {
+        count += mtc.second;
+      }
+    }
+    return count;
+  }
+
```

Comment:
I don't see why you chose to omit `kiwix::parseMimetypeCounter()`, providing this narrower-purpose template function in a header file, instead.

IMHO, taking into account that it is used for a single purpose, providing a non-template specific variant of it `countOfMediaMimetypes()` should be enough, which would result in a significantly simpler unit-test.

Currently your unit-test is too complex because it tests several things at once - parsing (which is the real source of complexity at the core of this functionality), filtering and summation (the latter two being quite trivial operations).

If you consider it as a general purpose utility that may be re-used in the future for another purpose, `kiwix::parseMimetypeCounter()` is a more general one and deserves being preserved.

Reply:
I omit `kiwix::parseMimeTypeCounter` to avoid the creation of a map where we just need to count the value.
But it is indeed early optimisation. It is better to separate the two concept. I will re add it.